play-framework/websockets/app/controllers/HomeController.java

```diff
@@ -0,0 +1,67 @@
+package controllers;
+
+import actors.Messenger;
+import akka.actor.ActorSystem;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Flow;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import com.fasterxml.jackson.databind.JsonNode;
+import dto.MessageDTO;
+import lombok.extern.slf4j.Slf4j;
+import play.libs.F;
+import play.libs.streams.ActorFlow;
+import play.mvc.*;
+import utils.MessageConverter;
+
+import javax.inject.Inject;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+@Slf4j
+public class HomeController extends Controller {
+    private final ActorSystem actorSystem;
+    private final Materializer materializer;
+
+    @Inject
+    public HomeController(ActorSystem actorSystem, Materializer materializer) {
+        this.actorSystem = actorSystem;
+        this.materializer = materializer;
+    }
+
+    public Result index(Http.Request request) {
+        String url = routes.HomeController.socket().webSocketURL(request);
+        //String url = routes.HomeController.akkaStreamsSocket().webSocketURL(request);
```

Comment:
Why is this commented out? If it's for the reader to experiment with a different option, then comment to them the reason for it being commented out.

Reply:
It was to give the user the ability to switch. So I have added the comments